ledger/eval_test.go

```diff
@@ -1402,3 +1410,57 @@ func TestModifiedAppLocalStates(t *testing.T) {
 		assert.False(t, created)
 	}
 }
+
+// TestAppInsMinBalance checks that accounts with MaxAppsOptedIn are accepted by block evaluator
+// and do not cause any MaximumMinimumBalance problems
+func TestAppInsMinBalance(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	genBalances, addrs, _ := newTestGenesis()
+	l := newTestLedger(t, genBalances)
+	defer l.Close()
+
+	const appid basics.AppIndex = 1
+
+	maxAppsOptedIn := config.Consensus[protocol.ConsensusFuture].MaxAppsOptedIn
+	require.Greater(t, maxAppsOptedIn, 0)
+	maxAppsCreated := config.Consensus[protocol.ConsensusFuture].MaxAppsCreated
+	require.Greater(t, maxAppsOptedIn, 0)
+
+	txnsCreate := make([]*txntest.Txn, 0, maxAppsOptedIn)
+	txnsOptIn := make([]*txntest.Txn, 0, maxAppsOptedIn)
+	appsCreated := make(map[basics.Address]int, len(addrs)-1)
+
+	acctIdx := 0
+	for i := 0; i < maxAppsOptedIn; i++ {
+		creator := addrs[acctIdx]
+		createTxn := txntest.Txn{
+			Type:              "appl",
+			Sender:            creator,
+			ApprovalProgram:   []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+			ClearStateProgram: []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+			Note:              randomNote(),
+		}
+		txnsCreate = append(txnsCreate, &createTxn)
+		count := appsCreated[creator]
+		count++
+		appsCreated[creator] = count
+		if count == maxAppsCreated {
+			acctIdx++
+		}
+
+		optInTxn := txntest.Txn{
+			Type:          "appl",
+			Sender:        addrs[9],
+			ApplicationID: appid + basics.AppIndex(i),
+			OnCompletion:  transactions.OptInOC,
+		}
+		txnsOptIn = append(txnsOptIn, &optInTxn)
+	}
+
+	eval := l.nextBlock(t)
+	txns := append(txnsCreate, txnsOptIn...)
+	eval.txns(t, txns...)
```

Comment:
Can this just be `eval.txns(txnsCreate..., txnsOptIn...)`

Reply:
no, the compiler complains
```
ledger/eval_test.go:1464:30: expected ')', found txnsOptIn
ledger/eval_test.go:1467:3: expected '}', found 'EOF'
```

Reply:
I guess ... must appear only once, at the end.